content/developers/dogstatsd/datagram_shell.md

```diff
@@ -0,0 +1,161 @@
+---
+
+
+This section specifies the raw datagram format for each data type that DogStatsD accepts. This isn't required reading if you're using any of the DogStatsD client libraries; however, if you want to write your own library, or use the shell to send metrics or events, then read on.
+
+## Datagram Format
+
+### Metrics
+
+`metric.name:value|type|@sample_rate|#tag1:value,tag2`
+
+- `metric.name` — a string with no colons, bars, or @ characters. See the [metric naming policy][1].
+- `value` — an integer or float.
+- `type` — `c` for counter, `g` for gauge, `ms` for timer, `h` for histogram, `s` for set.
+- `sample rate` (optional) — a float between 0 and 1, inclusive. Only works with counter, histogram, and timer metrics. Default is 1 (i.e. sample 100% of the time).
+- `tags` (optional) — a comma separated list of tags. Use colons for key/value tags, i.e. `env:prod`. The key `device` is reserved; Datadog drops a user-added tag like `device:foobar`.
+
+Here are some example datagrams:
+
+    # Increment the page.views counter
+    page.views:1|c
+
+    # Record the fuel tank is half-empty
+    fuel.level:0.5|g
+
+    # Sample the song length histogram half of the time
+    song.length:240|h|@0.5
+
+    # Track a unique visitor to the site
+    users.uniques:1234|s
+
+    # Increment the active users counter, tag by country of origin
+    users.online:1|c|#country:china
+
+    # Track active China users and use a sample rate
+    users.online:1|c|@0.5|#country:china
+
+### Events
+
+`_e{title.length,text.length}:title|text|d:timestamp|h:hostname|p:priority|t:alert_type|#tag1,tag2`
+
+- `_e` - The datagram must begin with `_e`
+- `title` — Event title.
+- `text` — Event text. Insert line breaks with an escaped slash (`\\n`)
+- `|d:timestamp` (optional) — Add a timestamp to the event. Default is the current Unix epoch timestamp.
+- `|h:hostname` (optional) - Add a hostname to the event. No default.
+- `|k:aggregation_key` (optional) — Add an aggregation key to group the event with others that have the same key. No default.
+- `|p:priority` (optional) — Set to 'normal' or 'low'. Default 'normal'.
+- `|s:source_type_name` (optional) - Add a source type to the event. No default.
+- `|t:alert_type` (optional) — Set to 'error', 'warning', 'info' or 'success'. Default 'info'.
+- `|#tag1:value1,tag2,tag3:value3...` (optional)— ***The colon in tags is part of the tag list string and has no parsing purpose like for the other parameters.*** No default.
+
+Here are some example datagrams:
+
+    # Send an exception
+    _e{21,36}:An exception occurred|Cannot parse CSV file from 10.0.0.17|t:warning|#err_type:bad_file
+
+    # Send an event with a newline in the text
+    _e{21,42}:An exception occurred|Cannot parse JSON request:\\n{"foo: "bar"}|p:low|#err_type:bad_request
+
+### Service Checks
+
+`_sc|name|status|d:timestamp|h:hostname|#tag1:value1,tag2,tag3:value3,...|m:service_check_message`
+
+- `_sc` — the datagram must begin with `_sc`
```

Comment:
Would a table layout be more readable here?

Reply:
I don't have a strong opinion either way. Imho we leave the list for now and switch it up later if we like.